src/Utilities/Compiler/Options/SymbolNamesWithValueOption.cs

```diff
@@ -373,16 +373,14 @@ internal TestAccessor(SymbolNamesWithValueOption<TValue> symbolNamesWithValueOpt
         /// </example>
         public sealed class NameParts
         {
-            public NameParts(string symbolName, TValue associatedValue = default)
+            public NameParts(string symbolName, TValue? associatedValue = default)
```

Comment:
From a code quality standpoint, it would be vastly preferable to remove the default value here.

```suggestion
            public NameParts(string symbolName, TValue associatedValue)
```

Most of the rest of the changes in this pull request are not necessary with this change.

Reply:
I will give it another try but I started with that and couldn't do the fix.

Reply:
I confirm that I am blocked, basically now [line 92](https://github.com/dotnet/roslyn-analyzers/pull/4543/files#diff-d26839098724560378df49fbfe7309653539ef2089e47babbee13a07e63ec988R92) we need a default value, so I need to pass it to the method signature and going up the chain I end up to places using `ISymbol` so I can't pass a value here. Note that in this case we pass the lambda so we should not use the default value.

Reply:
I'm going to do one more pass locally to experiment with this and see if I need to file a compiler feature request.

Reply:
Sure